### PR TITLE
Add friendrequest sys message && fix for new tslint rule

### LINKED ISSUE #260: add sys message in FriendRequest Event
According to the reply of JohnChang in Wechaty Developers' Home:

### Expected behavior

fire friendRequest event when accepted a friend request

### Actual behavior

cannot fire friendRequest event

### Fix

This bug because we cannot recognize this sys message, so wechaty cannot fire friendrequest event. I will add sys message in fire.ts then.

### sys message as follows:

> johnbassserver@gmail.com just added you to his/her contacts list. Send a message to him/her now!

If others have any sys message, please help to add in this issue, Thanks!

### PR DESCRIPTION
**Pull request by lijiarui**
tslint #251 

fix #260

### CODE REVIEW

**Comment by lijiarui** [comment]
I really regret what I had did, such a big work!

File: example/ding-dong-bot.ts [hunk 1]
```diff
@@ -65,8 +65,7 @@ bot
     const room = m.room()
     console.log((room ? '[' + room.topic() + ']' : '')
                 + '<' + m.from().name() + '>'
-                + ':' + m.toStringDigest()
-    )
+                + ':' + m.toStringDigest())
```
Comment (zixia) [inline example/ding-dong-bot.ts]: Please keep the original coding style.
Just add a comma to the end, and keep `)` at the new line.

File: example/gist-bot/on-message.ts [hunk 2]
```diff
@@ -18,8 +18,7 @@ export async function onMessage(message: Message): Promise<void> {
 
     console.log((room ? '[' + room.topic() + ']' : '')
                 + '<' + sender.name() + '>'
-                + ':' + message.toStringDigest()
-    )
+                + ':' + message.toStringDigest())
```
Comment (zixia) [inline example/gist-bot/on-message.ts]: Keep the original coding style of `)`

File: example/gist-bot/on-room-join.ts [hunk 3]
```diff
@@ -28,8 +28,7 @@ export async function onRoomJoin(
     if (room.topic() !== 'ding') {
       this.say('Room ' + room.topic()
             + ' got new memeber ' + inviteeName
-            + ' invited by ' + inviter.name()
-      )
+            + ' invited by ' + inviter.name())
```
Comment (zixia) [inline example/gist-bot/on-room-join.ts]: Do not move `)`

File: example/room-bot.ts [hunk 4]
```diff
@@ -110,8 +110,7 @@ bot
   log.info('Bot', 'EVENT: room-join - Room %s got new member %s, invited by %s'
                 , room.topic()
                 , inviteeList.map(c => c.name()).join(',')
-                , inviter.name()
-          )
+                , inviter.name())
```
Comment (zixia) [inline example/room-bot.ts]: keep `)`

File: example/room-bot.ts [hunk 5]
```diff
@@ -120,8 +119,7 @@ bot
 .on('room-leave', function(this, room, leaverList) {
   log.info('Bot', 'EVENT: room-leave - Room %s lost member %s'
                 , room.topic()
-                , leaverList.map(c => c.name()).join(',')
-              )
+                , leaverList.map(c => c.name()).join(','))
```
Comment (zixia) [inline example/room-bot.ts]: keep

File: example/room-bot.ts [hunk 6]
```diff
@@ -132,8 +130,7 @@ bot
     log.info('Bot', 'EVENT: room-topic - Room %s change topic to %s by member %s'
                   , oldTopic
                   , topic
-                  , changer.name()
-                )
+                  , changer.name())
```
Comment (zixia) [inline example/room-bot.ts]: keep

File: example/room-bot.ts [hunk 7]
```diff
@@ -149,8 +146,7 @@ bot
 
   console.log((room ? '[' + room.topic() + ']' : '')
               + '<' + sender.name() + '>'
-              + ':' + message.toStringDigest()
-  )
+              + ':' + message.toStringDigest())
```
Comment (zixia) [inline example/room-bot.ts]: keep

File: example/room-bot.ts [hunk 8]
```diff
@@ -256,8 +252,7 @@ function manageDingRoom() {
     room.on('join', function(this, inviteeList, inviter) {
       log.verbose('Bot', 'Room EVENT: join - %s, %s'
                         , inviteeList.map(c => c.name()).join(', ')
-                        , inviter.name()
-      )
+                        , inviter.name())
```
Comment (zixia) [inline example/room-bot.ts]: keep

File: example/room-bot.ts [hunk 9]
```diff
@@ -275,8 +270,7 @@ function manageDingRoom() {
       log.info('Bot', 'Room EVENT: topic - changed from %s to %s by member %s'
           , oldTopic
           , topic
-          , changer.name()
-        )
+          , changer.name())
```
Comment (zixia) [inline example/room-bot.ts]: keep the original position of `)`, just add a comma.

Will not comment on this question anymore.

File: src/contact.ts [hunk 10]
```diff
@@ -9,20 +9,20 @@ import { Wechaty }    from './wechaty'
...
+  province:   string,
+  alias:      string|null,
+  sex:        Gender,
+  signature:  string,
+  star:       boolean,
+  stranger:   boolean,
+  uin:        string,
+  weixin:     string,
+  avatar:     string,
+}  // XXX URL of HeadImgUrl
```
Comment (zixia) [inline src/contact.ts]: keep `avatar` and the comment stay in the same line.

File: src/contact.ts [hunk 11]
```diff
@@ -37,8 +37,7 @@ export type ContactRawObj = {
   UserName:     string
   HeadImgUrl:   string
 
-  stranger:     string // assign by injectio.js
-}
+  stranger:     string} // assign by injectio.js
```
Comment (zixia) [inline src/contact.ts]: please keep the original coding style.

File: src/io.ts [hunk 12]
```diff
@@ -24,8 +24,7 @@ export type IoSetting = {
   wechaty:    Wechaty
   token:      string
   apihost?:   string
-  protocol?:  string
-}
+  protocol?:  string}
```
Comment (zixia) [inline src/io.ts]: please keep your coding style consistent.

File: src/io.ts [hunk 13]
```diff
@@ -40,8 +39,7 @@ type IoEventName =  'botie'
 
 type IoEvent = {
   name:     IoEventName
-  payload:  any
-}
+  payload:  any}
```
Comment (zixia) [inline src/io.ts]: also here.

File: src/message.spec.ts [hunk 14]
```diff
@@ -102,16 +102,14 @@ test('ready()', async t => {
 
 test('find()', async t => {
   const m = await Message.find({
-    id: 'xxx'
-  })
+    id: 'xxx'})
```
Comment (zixia) [inline src/message.spec.ts]: and here

File: src/message.spec.ts [hunk 15]
```diff
@@ -102,16 +102,14 @@ test('ready()', async t => {
...
 
   t.truthy(m.id, 'Message.find')
 })
 
 test('findAll()', async t => {
   const msgList = await Message.findAll({
-    from: 'yyy'
-  })
+    from: 'yyy'})
```
Comment (zixia) [inline src/message.spec.ts]: here too

File: src/message.ts [hunk 16]
```diff
@@ -112,8 +112,7 @@ export type MsgRawObj = {
    * MsgType == CONF.MSGTYPE_SHARECARD" ng-click="showProfile($event,message.RecommendInfo.UserName)
    * MsgType == CONF.MSGTYPE_VERIFYMSG
    */
-  RecommendInfo?:   RecommendInfo
-}
+  RecommendInfo?:   RecommendInfo}
```
Comment (zixia) [inline src/message.ts]: ...

File: src/message.ts [hunk 17]
```diff
@@ -125,9 +124,8 @@ export type MsgObj = {
   status:   string
   digest:   string
   date:     string
-
-  url?:     string  // for MessageMedia class
-}
+  // for MessageMedia class
+  url?:     string}
```
Comment (zixia) [inline src/message.ts]: how can you make this happen?

File: src/message.ts [hunk 18]
```diff
@@ -136,10 +134,9 @@ export type MsgObj = {
 // export type MessageTypeValue = 1 | 3 | 34 | 37 | 40 | 42 | 43 | 47 | 48 | 49 | 50 | 51 | 52 | 53 | 62 | 9999 | 10000 | 10002
 
 export type MsgTypeMap = {
-  [index: string]: string|number
+  [index: string]: string|number}
   //   MessageTypeName:  MessageTypeValue
   // , MessageTypeValue: MessageTypeName
-}
```
Comment (zixia) [inline src/message.ts]: WTF...

File: src/puppet-web/friend-request.spec.ts [hunk 19]
```diff
@@ -13,9 +13,7 @@ import { Message }  from '../message'
 import { Puppet }   from '../puppet'
 import { FriendRequest as PuppetWebFriendRequest } from './friend-request'
 
-Config.puppetInstance({
-  userId: 'xxx'
-} as Puppet)
+Config.puppetInstance({ userId: 'xxx' } as Puppet)
```
Comment (zixia) [inline src/puppet-web/friend-request.spec.ts]: Keep the code in separate line, to make it more readable.

File: src/room.ts [hunk 20]
```diff
@@ -16,39 +16,34 @@ type RoomObj = {
   ownerUin:   number
   memberList: Contact[]
   nameMap:    Map<string, string>
-  aliasMap:   Map<string, string>
-}
+  aliasMap:   Map<string, string>}
```
Comment (zixia) [inline src/room.ts]: put `}` in new line.

File: src/room.ts [hunk 21]
```diff
@@ -16,39 +16,34 @@ type RoomObj = {
...
 
 type NameType = 'name' | 'alias'
 
 export type RoomRawMember = {
   UserName:     string
   NickName:     string
-  DisplayName:  string
-}
+  DisplayName:  string}
```
Comment (zixia) [inline src/room.ts]: new line

File: src/room.ts [hunk 22]
```diff
@@ -16,39 +16,34 @@ type RoomObj = {
...
 
 export type RoomRawObj = {
   UserName:         string
   EncryChatRoomId:  string
   NickName:         string
   OwnerUin:         number
   ChatRoomOwner:    string
-  MemberList:       RoomRawMember[]
-}
+  MemberList:       RoomRawMember[]}
```
Comment (zixia) [inline src/room.ts]: new line

File: src/room.ts [hunk 23]
```diff
@@ -16,39 +16,34 @@ type RoomObj = {
...
 export type RoomQueryFilter = {
-  topic: string | RegExp
-}
+  topic: string | RegExp}
 
 export type MemberQueryFilter = {
   name?:  string
-  alias?: string
-}
+  alias?: string}
```
Comment (zixia) [inline src/room.ts]: new line

File: src/room.ts [hunk 24]
```diff
@@ -176,8 +171,7 @@ export class Room extends EventEmitter implements Sayable {
                       , content
                       , Array.isArray(replyTo)
                         ? replyTo.map(c => c.name()).join(', ')
-                        : replyTo ? replyTo.name() : ''
-    )
+                        : replyTo ? replyTo.name() : '')
```
Comment (zixia) [inline src/room.ts]: put `)` in new line

File: src/room.ts [hunk 25]
```diff
@@ -333,8 +327,7 @@ export class Room extends EventEmitter implements Sayable {
       Config.puppetInstance().roomTopic(this, newTopic)
                               .catch(e => {
                                 log.warn('Room', 'topic(newTopic=%s) exception: %s',
-                                                  newTopic, e && e.message || e
-                                )
+                                                  newTopic, e && e.message || e)
```
Comment (zixia) [inline src/room.ts]: new line

File: src/room.ts [hunk 26]
```diff
@@ -406,8 +399,7 @@ export class Room extends EventEmitter implements Sayable {
     log.silly('Room', 'member({ %s })'
                         , Object.keys(queryArg)
                                 .map(k => `${k}: ${queryArg[k]}`)
-                                .join(', ')
-            )
+                                .join(', '))
```
Comment (zixia) [inline src/room.ts]: `)` and `}` in new line.
will not comment on this problem anymore.

File: src/util-lib.ts [hunk 27]
```diff
@@ -43,10 +43,10 @@ export class UtilLib {
     }
     return html
           .replace(/<img class="(\w*?emoji) (\w*?emoji[^"]+?)" text="(.*?)_web" src=[^>]+>/g
-                 , '$3'
+                 , '$3',
                  ) // <img class="emoji emoji1f4a4" text="[流汗]_web" src="/zh_CN/htmledition/v2/images/spacer.gif" />
           .replace(/<span class="(\w*?emoji) (\w*?emoji[^"]+?)"><\/span>/g
-                 , '[$2]'
+                 , '[$2]',
```
Comment (zixia) [inline src/util-lib.ts]: move `,` to the end of line.

File: src/util-lib.ts [hunk 28]
```diff
@@ -43,10 +43,10 @@ export class UtilLib {
     }
     return html
           .replace(/<img class="(\w*?emoji) (\w*?emoji[^"]+?)" text="(.*?)_web" src=[^>]+>/g
```
Comment (zixia) [inline src/util-lib.ts]: move `,` to the end of line.

File: src/util-lib.ts [hunk 29]
```diff
@@ -63,10 +63,10 @@ export class UtilLib {
...
-                 , '<emoji code="$2"/>'
+                 , '<emoji code="$2"/>',
```
Comment (zixia) [inline src/util-lib.ts]: move `,` to the end of line.
will not comment on this problem anymore.

**Pull request closed without merging**